python_modules/dagster/dagster/_core/execution/context/system.py

```diff
@@ -973,18 +988,22 @@ def fetch_external_input_asset_version_info(self) -> None:
 
         self._input_asset_version_info = {}
         for key in all_dep_keys:
-            self._fetch_input_asset_version_info(key)
+            self._fetch_input_asset_materialization_and_version_info(key)
         self._is_external_input_asset_version_info_loaded = True
 
-    def _fetch_input_asset_version_info(self, key: AssetKey) -> None:
+    def _fetch_input_asset_materialization_and_version_info(self, key: AssetKey) -> None:
         from dagster._core.definitions.data_version import (
             extract_data_version_from_entry,
         )
 
         event = self._get_input_asset_event(key)
         if event is None:
             self._input_asset_version_info[key] = None
+            self._upstream_asset_materialization_events[key] = None
         else:
+            self._upstream_asset_materialization_events[key] = (
```

Comment:
TODO move this into the non-partitioned asset section of this code block so that it's clear partitioned assets are treated differently